tools/toolchains/arm.py

```diff
@@ -71,7 +77,9 @@ def __init__(self, target, notify=None, macros=None,
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
 
-        if getattr(target, "default_toolchain", "ARM") == "uARM":
+        self.check_c_lib_supported(target, "arm")
+
+        if getattr(target, "default_toolchain", "ARM") == "uARM" or getattr(target, "default_lib", "std") == "small":
```

Comment:
```suggestion
        if (
            getattr(target, "default_toolchain", "ARM") == "uARM"
            or getattr(target, "default_lib", "std") == "small"
        ):
```
?

Reply:
One of your earlier review comments to use without brackets like below

 if getattr(target, "default_toolchain", "ARM") == "uARM"
    or getattr(target, "default_lib", "std") == "small":

```

Reply:
That cannot be true because the syntax is wrong and wont run.
The comment you are referring to is one where you had `if (condition):` and I said the brackets are not needed and that you should write `if condition:` which has a correct syntax.